packages/widget.vite-plugin.unstable/src/dev-plugin/FoundryWidgetDevPlugin.ts

```diff
@@ -64,6 +68,25 @@ export function FoundryWidgetDevPlugin(): Plugin {
       htmlEntrypoints = getInputHtmlEntrypoints(options);
     },
 
+    /**
+     * Capture configuration options from the Vite config. This is used to determine if we are running in
+     * Code Workspaces mode and in dev mode.
+     */
+    config(resolvedConfig, { command }) {
+      // Check for the optional `--mode` argument when running `vite`
+      isCodeWorkspacesMode = resolvedConfig.mode === CODE_WORKSPACES;
```

Comment:
It looks like all the places we pass `isCodeWorkspacesMode` to we also pass `server: ViteDevServer`. Do you prefer to do it this way to capture this once during config resolved, or reuse the existing `server` parameter and compute it on the fly so it doesn't need to be passed around?
```
isCodeWorkspacesMode(server: ViteServer) => server.config.mode === CODE_WORKSPACES
```